Resources/Prototypes/_NF/Entities/Mobs/NPCs/pets.yml

```diff
@@ -223,6 +223,106 @@
     factions:
       - Cat
 
+- type: entity
+  name: blood cult cat
+  parent: [NFMobPet, MobCatGhost]
+  id: MobCatBloodCult
+  description: Founded the Blood Cult out of boredoom.
+  components:
+  - type: Body
+    prototype: Bloodsucker
+    requiredLegs: 1
+  - type: MeleeWeapon
+    damage:
+      types:
+        Slash: 1
+        Bloodloss: 1
+  - type: Sprite
+    drawdepth: Mobs
+    sprite: _NF/Mobs/Pets/cat.rsi
+    layers:
+    - map: ["enum.DamageStateVisualLayers.Base"]
+      state: cultcat
+    - map: [ "enum.DamageStateVisualLayers.BaseUnshaded" ]
+      state: cultcat_unshaded
+      shader: unshaded
+  - type: DamageStateVisuals
+    states:
+      Alive:
+        Base: cultcat
+        BaseUnshaded: cultcat_unshaded
+      Critical:
+        Base: cultcat_dead
+        BaseUnshaded: mask_null
+      Dead:
+        Base: cultcat_dead
+        BaseUnshaded: mask_null
+  - type: GhostRole
+    name: ghost-role-information-cult-cat-name
+    description: ghost-role-information-cult-cat-description
+    rules: ghost-role-information-cult-cat-rules
+    prototype: CatBloodCult
+  - type: RandomMetadata
+    nameSegments: [ names_cat_cult ]
+  - type: AutoImplant
+    implants:
+    - MedicalTrackingImplant
+    - RedLightImplant
+    - DissonantWhispersImplant
+  - type: IntrinsicRadioTransmitter
+    channels:
+    - Science
+  - type: ActiveRadio
+    channels:
+    - Science
+  - type: NpcFactionMember
+    factions:
+      - Cat
+      - BloodCultNF
+  - type: NFSalvageMobRestrictions
+    despawnIfOffLinkedGrid: false
+    addComponentsOnDeath: []
+    removeComponentsOnDeath: []
+  - type: CargoSellBlacklist
+  - type: ActionGun
+    action: ActionEldritchPush
+    gunProto: EldritchPushGun
+
+- type: entity
+  id: EldritchPushGun
+  name: eldritch hand gun
+  description: Shoots eldritch hand.
+  categories: [ HideSpawnMenu ]
+  components:
+  - type: RechargeBasicEntityAmmo
+    rechargeCooldown: 300
+    rechargeSound:
+      path: /Audio/Magic/fireball.ogg
+  - type: BasicEntityAmmoProvider
+    proto: BulletEldritchHand
+    capacity: 1
+    count: 1
+  - type: Gun
+    soundGunshot:
+      path: /Audio/Effects/teleport_arrival.ogg
+    soundEmpty: null
+    projectileSpeed: 3
+
+- type: entity
+  id: ActionEldritchPush
+  categories: [ HideSpawnMenu ]
+  name: "[color=yellow]Force Push[/color]"
+  description: "Summon forth an eldritch hand. [color=yellow]3 minutes cooldown.[/color]"
+  components:
+  - type: WorldTargetAction
+    icon: { sprite : _NF/Effects/bloodcultprojectiles.rsi, state: hand }
+    event: !type:ActionGunShootEvent
+    priority: 2
+    checkCanAccess: false
+    range: 0
+    itemIconStyle: BigAction
+    useDelay: 300
```

Comment:
The delay here doesn't seem to be working on my end, I'd expect a cooldown ring, but the baby dragon's also isn't working.  Might've been broken, unsure why.

Reply:
Yeah, it doesn't

Reply:
Alrighty, noted.